File: DataTier1/Data/Models/Post.cs
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;

namespace Data.Models;

public class Post : BaseEntity
{
    [Required]
    public string Title { get; set; }
    [Required]
    public string Content { get; set; }
    [Required]
    public User Author { get; set; }
    [Required]
    public Guid AuthorId { get; set; }
    public Event Event { get; set; }
    public Guid EventId { get; set; }

    [Required]
    public string CreationDate { get; set; }
    
}

File: CafeTier2/src/main/java/group7/Restful/entity/Post.java
package group7.Restful.entity;

import lombok.Data;

import java.util.UUID;
@Data
public class Post {
    private UUID id;
    private String title;
    private String content;
    private String author;
    private String event;

    public Post() {
    }
    public Post(UUID id, String title, String content, String author) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.author = author;
    }
}
